frame/support/src/lib.rs

```diff
@@ -1381,15 +1381,23 @@ pub mod pallet_prelude {
 /// alias then the generation of the getter might fail. In this case getter can be implemented
 /// manually.
 ///
+/// NOTE: The generic `Hasher` must implement the [`StorageHasher`] trait (or the type is not
+/// usable at all). We use [`StorageHasher::METADATA`] for the metadata of the hasher of the
+/// storage item. Thus generic hasher is supported.
+///
 /// ### Macro expansion
 ///
 /// For each storage the macro generate a struct named
 /// `_GeneratedPrefixForStorage$NameOfStorage`, implements `StorageInstance` on it using pallet
 /// name and storage name. And use it as first generic of the aliased type.
 ///
 ///
-/// The macro implement the function `storage_metadata` on `Pallet` implementing the metadata for
-/// storages.
+/// The macro implements the function `storage_metadata` on `Pallet` implementing the metadata for
+/// storage types:
```

Comment:
@dvdplm Another alternative would be "storage kinds" I guess? The way I read this it talks about the different kinds of storage items you can define (namely values, maps and double maps), not the particular storage items (which for me are the actual instances).
```suggestion
/// the different storage kinds:
```

Reply:
actually the macro implements the function `storage_metadata`, which holds the metadata for every instance of storage.

I was more thinking here about the instance rather than the kind

Reply:
closed in favor of your other suggestion :-)